conjure-java-undertow-runtime/src/main/java/com/palantir/conjure/java/undertow/runtime/EndpointHandlerWrapper.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.conjure.java.undertow.runtime;
+
+import com.palantir.conjure.java.undertow.lib.Endpoint;
+import io.undertow.server.HandlerWrapper;
+
+public interface EndpointHandlerWrapper {
+
+    /** This should return the {@link HandlerWrapper} that should be applied to the {@link Endpoint#handler()}. **/
+    HandlerWrapper wrap(Endpoint endpoint);
```

Comment:
No reason for one wrapper to return a second wrapper.
```suggestion
    HttpHandler wrap(Endpoint endpoint);
```

Reply:
I think it's actually better with an `HandlerWrapper` because then what are actually returning is a "Filter" instead of a "Filtered Handler".

Reply:
I don't agree, the `Endpoint` is a single instance of a single endpoint, there's no reason to wrap multiple handlers with the same wrapper. Returning the handler directly results in an API that's easier to understand.

Reply:
I don't agree but I did it anyway